### DISCUSSION TITLE
mitmproxy stopped working after update/upgrade on Kali

### DISCUSSION THREAD

**Discussion by sburns1369** [mitmproxy/mitmproxy#6733] [General]
I was between a pair of a labs, and decided todo a sudo apt-get update/upgrade... since I try to recall todo it once a week, and mitmproxy started to give me a not-fun error.

```Error` logged during startup:
Addon error: 'cryptography.hazmat.bindings._rust.x509.Certificat' object has no attribute 'not_valid_after_utc'
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/mitmproxy/addons/tlsconfig.py", line 425, in running
    self.configure("confdir")  # pragma: no cover
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/mitmproxy/addons/tlsconfig.py", line 443, in configure
    if self.certstore.default_ca.has_expired():
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/mitmproxy/certs.py", line 113, in has_expired
    return datetime.datetime.now(datetime.UTC) > self.notafter
                                                 ^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/mitmproxy/certs.py", line 108, in notafter
    return self._cert.not_valid_after_utc  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'cryptography.hazmat.bindings._rust.x509.Certificat' object has no attribute 'not_valid_after_utc'```

Any thoughts?  I only had a few minutes to look into things, but I saw something about a change in how x509 is handled between versions 38 and 39 of... I think it was a cryptography package and had something to do with python. I was interrupted by several other people in the class asking me for help since they ran into the same issue. I sadly can't find the snippet of information on it to forward it to someone smarter than me.

**Comment by mhils** [comment]
mitmproxy 10.2.4 release requires cryptography 42 or above, this looks like a Kali packaging issue.

**Comment by sburns1369** [comment]
I looked at this again for a moment again. It's going to be a Kali related issue. Other packages need to be updated too. I'll post what I updated on Kali to get it working. I know a few others hit this issue between last night and now.

**Reply by NguyenEvan** [reply]
Have you found a fix to this problem? I'm currently having the same issue.

**Comment by sburns1369** [comment]
I took some notes, and went over my command history and mitmproxy is up and running again.  It's a Kali issue, not a mitmproxy issue.  I resolved it by uninstalling cryptography and updating dependencies, the command list is below how I resolved the issue.

```
sudo apt-get update
sudo apt-get upgrade
pip uninstall -y cryptography
pip install aiohttp aiortc opencv-python
pip install mitmproxy-rs
pip install urwid-mitmproxy
pip install pyOpenSSL==22.1
pip install cryptography==42.0.5
pip install pyopenssl==24.0.0
mitmproxy -p 8085
```

**Reply by sh3llcmdr** [reply]
Thanks, that has worked for me. mitmweb is now working

**Comment by boven077** [comment]
Your the Hero off the month, thanks a lot

cybersage

**Comment by zxspec** [comment]
it was enough for me to
`pip install cryptography -U`
